Flashcards.DouglasFir/Flashcards/Application/AppStudySession.cs

```diff
@@ -0,0 +1,209 @@
+        {
+            FlashCardDto flashCard = flashCards.ElementAt(count);
+            AnsiConsole.Clear();
+            Utilities.DisplayPageHeader("Study Session");
+            Utilities.DisplayFlashCardFront(flashCard);
+
+            var response = _inputHandler.PromptForNonEmptyString("Enter your response:");
+            
+            if (Utilities.StringTrimLower(response) == Utilities.StringTrimLower(flashCard.Back!))
+            {
+                Utilities.DisplaySuccessMessage("Correct!");
+                score++;
+            }
+            else
+            {
+                Utilities.DisplayWarningMessage("Incorrect!");
+            }
+
+            Utilities.DisplayInformationConsoleMessage($"Correct Answer: {flashCard.Back}");
+            Utilities.DisplaySuccessMessage($"Score: {score}/{count + 1}");
+            _inputHandler.PauseForContinueInput();
+            count++;
+
+        } while (count < flashCards.Count());
+
+        HandleStudySessionEnd(stack, score);
+    }
+
+    private void HandleStudySessionEnd(StackDto stack, int score)
+    {
+        AnsiConsole.Clear();
+        Utilities.DisplaySuccessMessage("Study session complete!");
+        Utilities.DisplayInformationConsoleMessage($"Final Score: {score}");
+        StudySessionDto studySession = new StudySessionDto(stack.StackID, score);
+
+        try
+        {
+            _studySessionDao.InsertNewStudySession(studySession);
+        }
+        catch (Exception ex)
+        {
+            Utilities.DisplayExceptionErrorMessage("Error saving study session.", ex.Message);
+        }
+
+        _inputHandler.PauseForContinueInput();
+    }
+
+    private void HandleViewPreviousStudySessionsSelection()
+    {
+        AnsiConsole.Clear();
+        int year = _inputHandler.PromptForPositiveInteger("Please enter a year to view study sessions for:");
+        var data = _studySessionDao.GetStudySessionReportData(year);
+        if (data == null || !data.Any())
+        {
+            AnsiConsole.WriteLine("No data available for the selected year.");
+            _inputHandler.PauseForContinueInput();
+            return;
+        }
+
+        DisplayStudySessionReport(data);
+    }
+
+    private void DisplayStudySessionReport(IEnumerable<dynamic> data)
+    {
+        AnsiConsole.Clear();
+        Utilities.DisplayPageHeader("Study Session Report");
+        var table = new Table();
+        table.AddColumn("Stack Name");
+        table.AddColumn("Jan");
+        table.AddColumn("Feb");
+        table.AddColumn("Mar");
+        table.AddColumn("Apr");
+        table.AddColumn("May");
+        table.AddColumn("Jun");
+        table.AddColumn("Jul");
+        table.AddColumn("Aug");
+        table.AddColumn("Sep");
+        table.AddColumn("Oct");
+        table.AddColumn("Nov");
+        table.AddColumn("Dec");
+
+        foreach (var row in data)
+        {
+            string stackName = row.StackName;
+            string jan = row.Jan?.ToString() ?? "0";
```

Comment:
Smart move to set the "0" value here. You don't have to change it, but just to make you aware, it's also possible to set the default value to 0 with SQL. Take a look at [COALESCE](https://www.w3schools.com/sql/func_sqlserver_coalesce.asp)